### ISSUE TITLE
[Bug] Image format extensions messed up

### ISSUE THREAD

**Issue by Paritosh-Negi** [abhijithvijayan/ghost-on-github-pages#25]
Demo URL: https://paritosh-negi.github.io/myghostblog/

Images are not loading because of wrong image format extensions.

Can anyone help me to fix this?

**Comment by abhijithvijayan** [comment]
@Paritosh-Negi I will look onto it asap.

**Comment by Paritosh-Negi** [comment]
Yes, I followed the instructions carefully twice; still, images are not loading.

**Comment by abhijithvijayan** [comment]
@Paritosh-Negi  can reproduce the issue, I guess it is the issue with buster tool.

Let me see if I can manually fix that with some changes to the script.

**Comment by abhijithvijayan** [comment]
@Paritosh-Negi I have fixed the issue. Delete the existing `deploy.sh` script and replace with new content from https://raw.githubusercontent.com/abhijithvijayan/ghost-on-github-pages/master/deploy.sh

Give execute permission(one time only)

`chmod u+x deploy.sh`

Then deploy with
`
./deploy.sh
`

Feel free to reopen the issue if this doesn't solve your issue.

**Comment by Paritosh-Negi** [comment]
Thank you, all images are loading now 👍

**Comment by paddy420Smokers** [comment]
Hi, Im still getting the error.

**Comment by abhijithvijayan** [comment]
Did you use the new script?

**Comment by abhijithvijayan** [comment]
Also do create a demo repo

**Comment by paddy420Smokers** [comment]
Yes Ive used the new script. Do I need a demo repo for it to work? Anyway thanks for your quick reply. Thought this thread was dead.

**Comment by paddy420Smokers** [comment]
Heres the Demo Ive setup: https://paddy420smokers.github.io/cannalogie/ (You can see that the first changed image is not appearing.)

Git Rep: https://github.com/paddy420Smokers/cannalogie 

Note: Ive changed the port from 2368 (Its already in use) to 2380. Ive changed the port in the new deploy.sh and in the development.js did I miss something? 

Also Im struggling with the Image bug for a few weeks now. I hope you can help me! Thanks again for your great work :)

**Comment by paddy420Smokers** [comment]
<img width="1440" alt="Bildschirmfoto 2019-08-19 um 12 12 14" src="https://user-images.githubusercontent.com/39915222/63253655-ed077b80-c27a-11e9-8c49-542591ee5070.png">

I can see that something messes up with the Image ending 

Check the attached picture.

**Comment by abhijithvijayan** [comment]
@paddy420Smokers I will look on it soon

**Comment by tejakummarikuntla** [comment]
Hi @abhijithvijayan ,
I'm facing the exact issue, Any updates to this solution?

**Comment by abhijithvijayan** [comment]
This is an easy fix.

I didnt get any updates on this, hence I missed.

**Comment by tejakummarikuntla** [comment]
No @abhijithvijayan , I didn't see its fixed.
The image extensions are getting messed up still.

**Comment by abhijithvijayan** [comment]
Will get this fixed asap.

**Comment by tejakummarikuntla** [comment]
No @abhijithvijayan , I didn't see its fixed.
The image extensions are getting messed up still.

I don't think the issue is with `Buster` ,
I tried doing the same thing with Wget, [I didn't use Buster here]
```wget -r -nH -P docs -E -T 2 -np -k http://localhost:2368/
```
[Ref](https://zzamboni.github.io/test-ghost-blog/hosting-a-ghost-blog-in-github-the-easier-way/index.html)
Still, I find the same issue coming up, probably the issue is due to the latest update of Ghost.

**Comment by abhijithvijayan** [comment]
The deployment script here has fix for png files, all it takes is same code for jpg 

If you could attach the html image broken source here, I can attach the resolution as well

**Comment by abhijithvijayan** [comment]
Also thanks for suggesting way to use wget

I will migrate the instructions and script and fix all that issue and let you know

**Comment by tejakummarikuntla** [comment]
No @abhijithvijayan , I didn't use any JPG files, all I used is PNG.
Please find the attachment.

![png_extension_issue](https://user-images.githubusercontent.com/34749692/80316703-ed4e1280-881c-11ea-8b55-0b0e988b5d72.PNG)

**Comment by tejakummarikuntla** [comment]
This filename issue is occurring at three diff location which reflects at `post-card-image` image class.
1.  `static/index.html`
2.  `static/PUBLISHED_BLOG_TITLE/index.html
3.  `static/tag/CORRESPONDING_PUBLISHED_BLOG_TAG/index.html`

**Comment by abhijithvijayan** [comment]
@tejakummarikuntla 

As this issue was raised once before, I pushed a fix to resolve this, it was resolved to jpg images only.

As now with png, its a simple fix. All it takes is copying these lines as that of png

https://github.com/abhijithvijayan/ghost-on-github-pages/blob/682acca45af056b58da4b2e83964a05d7b2c6b3d/deploy.sh#L52

**Comment by tejakummarikuntla** [comment]
Hi @abhijithvijayan ,
Adding the following lines which mimics the .jpg format in deploy.sh, fixed this!
```
find static -name *.html -type f -exec sed -i '''s#.pngg 600w#''.png 600w''#g' {} \;		
find static -name *.html -type f -exec sed -i '''s#.pngng 1000w#''.png 1000w''#g' {} \;
find static -name *.html -type f -exec sed -i '''s#.pngpng 2000w#''.png 2000w''#g' {} \;
```

**Comment by abhijithvijayan** [comment]
Could you send a PR for this?

**Comment by tejakummarikuntla** [comment]
Absolutely !!!!

**Comment by abhijithvijayan** [comment]
@tejakummarikuntla Thanks.

Just one thing, buster generated urls as per user, does wget breaks the asset links or something without doing so?

**Comment by tejakummarikuntla** [comment]
But @abhijithvijayan , I think we gotta another problem.
The mobile view from chrome is totally fine, but the entire webpage is totally messed up when I open the site on my mobile.
- I faced this issue even before the .png fix [Only for this (casper (default) theme)]
Any possible soluntion?

**Comment by abhijithvijayan** [comment]
@tejakummarikuntla Could you attach the live link of the site?

I will take a look at it.
Probably wget broke something.

**Comment by tejakummarikuntla** [comment]
@abhijithvijayan ,
I tried using the png, jpg typo clearing bash script on `Wget` generated files instead of `Buster` generated files. It worked really well, I don't see any issues with the Mobile view now.
As we can see various issues are rising with `Buster` which requires py2 dependency as well.

Why don't we use `Wget` in this project instead of `Buster`. What do you say?

**Comment by abhijithvijayan** [comment]
Sure. I will accept PRs

**Comment by abhijithvijayan** [comment]
@tejakummarikuntla 
![Screenshot_20200427_013736](https://user-images.githubusercontent.com/34790378/80318444-ccd78580-8827-11ea-8b9e-72a6657e0bd7.png)

Your build is not successful with patched links. I will fix the script.

**Comment by abhijithvijayan** [comment]
@tejakummarikuntla Everything is set to use `wget`

The script fixes `jpeg`, `jpg`, `png` image formats.

Everything seems to work flawlessly. 

Try again with the latest script and feel to reopen this issue, if anything comes up.

**Comment by DieYiSu** [comment]
Hi @abhijithvijayan , 

I have found some problem when I run your deploy.sh on my Macbook Pro, and I have fixed it :)

- The sed shell command which trying to replace jpgg with jpg
- The sed shell symbol in the command which trying to replace localhost:2368 with username and gh_repo.
- The Fixing Images step should be to do on the setup step.
- The master branch had been replaced with main branch by Github team at 2020/10, so you don't need to create master and gh-pages branch.

I will send the PR to you.
